clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1105,6 +1105,20 @@ static ParmVarDecl *getSyclKernelHandlerArg(FunctionDecl *KernelCallerFunc) {
   return (KHArg != KernelCallerFunc->param_end()) ? *KHArg : nullptr;
 }
 
+static void getAccessModeReadVal(const TemplateArgument AccessModeArg,
+                                 llvm::APSInt &Val) {
```

Comment:
Why the 'out' parameter rather than just returning E->GetInitVal?

Reply:
Returned bool instead.